specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json

```diff
@@ -0,0 +1,9155 @@
+          "type": "integer",
+          "readOnly": true
+        }
+      }
+    },
+    "SelfHostedIntegrationRuntimeStatusTypeProperties": {
+      "type": "object",
+      "properties": {
+        "createTime": {
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        },
+        "taskQueueId": {
+          "type": "string",
+          "readOnly": true
+        },
+        "internalChannelEncryption": {
+          "type": "string",
+          "readOnly": true
+        },
+        "version": {
+          "type": "string",
+          "readOnly": true
+        },
+        "nodes": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SelfHostedIntegrationRuntimeNode"
+          }
+        },
+        "scheduledUpdateDate": {
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        },
+        "updateDelayOffset": {
+          "type": "string",
+          "readOnly": true
+        },
+        "localTimeZoneOffset": {
+          "type": "string",
+          "readOnly": true
+        },
+        "capabilities": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          },
+          "readOnly": true
+        },
+        "serviceUrls": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "readOnly": true
+        },
+        "autoUpdate": {
+          "enum": [
+            "On",
+            "Off"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "IntegrationRuntimeAutoUpdate",
+            "modelAsString": true
+          }
+        },
+        "versionStatus": {
+          "type": "string",
+          "readOnly": true
+        },
+        "pushedVersion": {
+          "type": "string",
+          "readOnly": true
+        },
+        "latestVersion": {
+          "type": "string",
+          "readOnly": true
+        },
+        "autoUpdateETA": {
+          "format": "date-time",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "SelfHostedIntegrationRuntimeStatusProperties": {
+      "type": "object",
+      "properties": {
+        "typeProperties": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/SelfHostedIntegrationRuntimeStatusTypeProperties"
+            }
+          ]
+        },
+        "state": {
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "SelfHostedIntegrationRuntimeStatus": {
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/IntegrationRuntimeStatus"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "allOf": [
+            {
+              "$ref": "#/definitions/SelfHostedIntegrationRuntimeStatusProperties"
+            }
+          ]
+        }
```

Comment:
Many of them should be a direct ref to the type, not an anonymous type with "allOf".

Swagger appears to be generated?

Reply:
Yes, this is autogenerated. Also, explained in a separate email.